relayer/src/index.ts

```diff
@@ -1,54 +1,46 @@
-import * as dotenv from "dotenv";
 import { ApiPromise, WsProvider } from "@polkadot/api";
-import { concatMap } from "rxjs/operators";
-import { getAccount } from "./account";
-
-dotenv.config();
+import { RegistryTypes } from "@polkadot/types/types";
 
-const sourceProvider = new WsProvider(process.env.SOURCE_CHAIN_URL);
-const targetProvider = new WsProvider(process.env.TARGET_CHAIN_URL);
+import { getAccount } from "./account";
+import config from "./config";
+import Source from "./source";
+import Target from "./target";
 
 // TODO: use typedefs from subspace.js
 const types = {
   PutDataObject: "Vec<u8>",
 };
 
-// TODO: remove IIFE when Eslint is updated to v8.0.0 (will support top-level await)
-(async () => {
-  const sourceApi = await ApiPromise.create({
-    provider: sourceProvider,
-  });
+const createApi = async (url?: string, types?: RegistryTypes) => {
+  if (!url) throw new Error("Endpoint url is not provided");
```

Comment:
Why is URL optional if API can't work without it?

Reply:
`process.env.TARGET_CHAIN_URL` is `string | undefined`

Reply:
Then validate whether URL has a sane value before calling this function.

I typically have a `Config` class in apps that holds all configuration necessary in it. If there are any required environment variables missing or contain the wrong value `Config` instantiation would fail early and application would crash loudly instead of silently sending wrong value all the way into this function.

Inputs should be checked as early as possible.